olp-cpp-sdk-core/src/http/android/NetworkAndroid.cpp

```diff
@@ -617,52 +640,48 @@ jobjectArray NetworkAndroid::CreateExtraHeaders(
-  jobjectArray headers = (jobjectArray)env->NewObjectArray(
+  const size_t extra_headers_count = 1;
+  const size_t header_count = (headers.size() + extra_headers_count) * 2;
+  jobjectArray jheaders = (jobjectArray)env->NewObjectArray(
       header_count, gStringClass, jempty_string);
-  if (!headers || env->ExceptionOccurred()) {
+  if (!jheaders || env->ExceptionOccurred()) {
     env->DeleteLocalRef(jempty_string);
     EDGE_SDK_LOG_ERROR(kLogTag, "Failed to create string array for headers");
     return 0;
   }
   env->DeleteLocalRef(jempty_string);
 
-  for (size_t i = 0; i < extra_headers.size(); ++i) {
-    jstring name = env->NewStringUTF(extra_headers[i].first.c_str());
-    if (!name || env->ExceptionOccurred()) {
-      EDGE_SDK_LOG_ERROR(kLogTag,
-                         "Failed to create extra header name string: index="
-                             << i << "; name=" << extra_headers[i].first);
+  for (size_t i = 0; i < headers.size(); ++i) {
+    if (!SetStringArrayElement(env, jheaders, i * 2, headers[i].first)) {
+      EDGE_SDK_LOG_ERROR(kLogTag, "Failed to create header string: index="
+                                      << i << "; name=" << headers[i].first);
       return 0;
     }
-    utils::JNIScopedLocalReference name_ref(env, name);
 
-    jstring value = env->NewStringUTF(extra_headers[i].second.c_str());
-    if (!value || env->ExceptionOccurred()) {
-      EDGE_SDK_LOG_ERROR(kLogTag,
-                         "Failed to create extra header value string: index="
-                             << i << "; value=" << extra_headers[i].second);
+    if (!SetStringArrayElement(env, jheaders, i * 2 + 1, headers[i].second)) {
+      EDGE_SDK_LOG_ERROR(kLogTag, "Failed to create header string: index="
+                                      << i << "; value=" << headers[i].second);
       return 0;
     }
-    utils::JNIScopedLocalReference value_ref(env, value);
+  }
 
-    env->SetObjectArrayElement(headers, i * 2, name);
-    if (env->ExceptionOccurred()) {
-      EDGE_SDK_LOG_ERROR(kLogTag,
-                         "Failed to set extra header value string: index="
-                             << i << "; name=" << extra_headers[i].first);
-      return 0;
-    }
-    env->SetObjectArrayElement(headers, i * 2 + 1, value);
-    if (env->ExceptionOccurred()) {
-      EDGE_SDK_LOG_ERROR(
-          kLogTag, "Failed to set extra header value string: index="
-                       << i << "; value=" << extra_headers[i].second.c_str());
-      return 0;
-    }
+  // Setup extra headers:
+  if (!SetStringArrayElement(env, jheaders, headers.size() * 2,
+                             kUserAgentHeader)) {
+    EDGE_SDK_LOG_ERROR(kLogTag, "Failed to create extra header string: index="
+                                    << headers.size() * 2
+                                    << "; name=" << kUserAgentHeader);
+    return 0;
+  }
+  if (!SetStringArrayElement(env, jheaders, headers.size() * 2 + 1,
```

Comment:
you need to check somehow if the user does not already has provided a similar header.
If this is the case then you probably need to merge them instead of providing duplicates. As far as I know not all HTTP headers allow duplicate entries.

Reply:
I checked that case on Android - this header will be merged automatically in the format:
User-Agent: <old_user_agent_value>; <new_user_agent_value>

Reply:
Please add a comment then so that we know that the headers will be merged

Reply:
Not valid anymore